apps/src/lib/node/ledger/mod.rs

```diff
@@ -58,6 +61,34 @@ const ENV_VAR_RAYON_THREADS: &str = "ANOMA_RAYON_THREADS";
 //```
 
 impl Shell {
+    fn load_proposals(&mut self) {
+        let proposals_key = gov_storage::get_commiting_proposals_prefix(
+            self.storage.last_epoch.0,
+        );
+
+        let (proposal_iter, _) = self.storage.iter_prefix(&proposals_key);
+        for (key, _, _) in proposal_iter {
+            let key =
+                Key::from_str(key.as_str()).expect("Key should be parsable");
+            if gov_storage::get_commit_epoch(&key).unwrap()
+                != self.storage.last_epoch.0
+            {
+                // NOTE: `iter_prefix` iterate over the matching prefix. In this
+                // case  a proposal with grace_epoch 110 will be
+                // matched by prefixes  1, 11 and 110. Thus we
+                // have to skip to the next iteration of
+                //  the cycle for all the prefixes that don't actually match
+                //  the desired epoch.
```

Comment:
oh, this is unfortunate - I wonder if we should instead write the proposal IDs with a fixed length string with zeroes in the prefix (maybe 20 chars to fit u64::MAX)

Reply:
its a little bit ugly but having a fixed size string would complicate how we handle the counter/proposal_id. I think this is fine for now